views/client.go

```diff
@@ -90,3 +191,141 @@ func startViewsQuery(playbackID, playbackRecordingID string) string {
 		queryID,
 	)
 }
+
+var viewershipBreakdownFields = map[string]string{
+	"device":      "device",
+	"os":          "os",
+	"user_agent":  "user_agent_family",
+	"continent":   "playback_continent_name",
+	"country":     "playback_country_name",
+	"subdivision": "playback_subdivisions_name",
+	"timezone":    "playback_timezone",
+	"viewer_id":   "viewer_id",
+}
+
+var allowedTimeSteps = map[string]bool{
+	"hour":  true,
+	"day":   true,
+	"week":  true,
+	"month": true,
+	"year":  true,
+}
+
+type LDSMStateRow struct {
+	TimeInterval time.Time `bigquery:"time_interval"`
+	PlaybackID   string    `bigquery:"playback_id"`
+
+	// breakdown fields
+	Device      string `bigquery:"device"`
+	OS          string `bigquery:"os"`
+	UserAgent   string `bigquery:"user_agent_family"`
+	Continent   string `bigquery:"playback_continent_name"`
+	Country     string `bigquery:"playback_country_name"`
+	Subdivision string `bigquery:"playback_subdivisions_name"`
+	TimeZone    string `bigquery:"playback_time_zone"`
+
+	// metric data
+	ViewCount  int64 `bigquery:"view_count"`
+	PlaytimeMs int64 `bigquery:"playtime_ms"`
+}
+
+func (c *Client) doViewsBigQuery(ctx context.Context, spec QuerySpec) ([]LDSMStateRow, error) {
+	query := squirrel.Select(
+		"countif(play_intent) as view_count",
+		"sum(playtime_ms) as playtime_ms",
```

Comment:
```suggestion
		"sum(playtime_ms) / 60000 as playtime_mins",
```
milliseconds is not interpretable. we should do minutes or hours. i think for the historical, minutes are best. the summary table uses hours but the bins can be small in this context and .25 hours is tough to interpret but 15 minutes is nice

Reply:
Yeah I love that